### ISSUE TITLE
Unable to run test worlds for underwater simulation

### ISSUE THREAD

**Issue by sauk2** [gazebosim/gz-sim#2344]
## Environment
* OS Version: Ubuntu 22.04
* Build: Source build
* Branch: gz-sim8 (commit: 5944188294511a60b829f88851edeaded0b72a84)
 
## Description

The test worlds for `DopplerVelocityLogSystem` plugin show a blank screen after launching. Specifically these files:
- [`test/worlds/bottomless_pit.sdf`](https://github.com/gazebosim/gz-sim/blob/gz-sim8/test/worlds/bottomless_pit.sdf)
- [`test/worlds/flat_seabed.sdf`](https://github.com/gazebosim/gz-sim/blob/gz-sim8/test/worlds/flat_seabed.sdf)
- [`test/worlds/underwater_currents.sdf`](https://github.com/gazebosim/gz-sim/blob/gz-sim8/test/worlds/underwater_currents.sdf)

## Steps to reproduce

The gz command is run with verbosity 2

```bash
  gz sim underwater_currents.sdf --verbose 2
```

#### This is the screen that is seen. No change occurs after this.

![Screenshot from 2024-03-28 01-10-14](https://github.com/gazebosim/gz-sim/assets/75178156/1a41bbe2-3e36-4150-a1e2-6dbed1887087)

#### Following is the generated log

```bash
[Wrn] [SdfEntityCreator.cc:800] Gazebo does not support Ogre material scripts. See https://gazebosim.org/api/sim/8/migrationsdf.html#:~:text=Materials for details.
[Wrn] [SdfEntityCreator.cc:805] Custom material scripts are not supported.
[Wrn] [SdfEntityCreator.cc:800] Gazebo does not support Ogre material scripts. See https://gazebosim.org/api/sim/8/migrationsdf.html#:~:text=Materials for details.
[Wrn] [SdfEntityCreator.cc:805] Custom material scripts are not supported.
[Wrn] [SdfEntityCreator.cc:800] Gazebo does not support Ogre material scripts. See https://gazebosim.org/api/sim/8/migrationsdf.html#:~:text=Materials for details.
[Wrn] [SdfEntityCreator.cc:805] Custom material scripts are not supported.
[Wrn] [SdfEntityCreator.cc:800] Gazebo does not support Ogre material scripts. See https://gazebosim.org/api/sim/8/migrationsdf.html#:~:text=Materials for details.
[Wrn] [SdfEntityCreator.cc:805] Custom material scripts are not supported.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:52:3: QML RenderWindowOverlay: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:67:3: QML EntityContextMenu: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:52:3: QML RenderWindowOverlay: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:67:3: QML EntityContextMenu: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:52:3: QML RenderWindowOverlay: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:67:3: QML EntityContextMenu: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:52:3: QML RenderWindowOverlay: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:67:3: QML EntityContextMenu: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:52:3: QML RenderWindowOverlay: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:67:3: QML EntityContextMenu: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:52:3: QML RenderWindowOverlay: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:67:3: QML EntityContextMenu: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityTree/EntityTree.qml:148:7: QML ToolButton: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityTree/EntityTree.qml:148:7: QML ToolButton: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/WorldStats/WorldStats.qml:53:3: QML RowLayout: Binding loop detected for property "x"
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:52:3: QML RenderWindowOverlay: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
[GUI] [Wrn] [Application.cc:908] [QT] file::/EntityContextMenuPlugin/EntityContextMenuPlugin.qml:67:3: QML EntityContextMenu: Detected anchors on an item that is managed by a layout. This is undefined behavior; use Layout.alignment instead.
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
[Wrn] [Ogre2Camera.cc:421] Ogre2Camera::SetVisibilityMask: Mask bits c0000000 are set but will be ignored as they conflict with the reserved bits used internally by the ogre2 backend.
```

**Comment by sauk2** [comment]
@azeey I noticed this while working on issue #2289

**Comment by azeey** [comment]
Hi @sauk2 , those files are meant to be used for integration tests. They don't need the `SceneBroadcaster` because the tests don't need the GUI. Adding it will unnecessarily increase test time. Most of the SDFormat files there intentionally don't have the `SceenBroadcaster` or other systems that are normally essential. Instead of your fix in #2346, what we need is a README in the `test/worlds` directory to tell users not to use these files, but instead refer to the `examples` directory.

**Comment by sauk2** [comment]
> Hi @sauk2 , those files are meant to be used for integration tests. They don't need the `SceneBroadcaster` because the tests don't need the GUI. Adding it will unnecessarily increase test time. Most of the SDFormat files there intentionally don't have the `SceenBroadcaster` or other systems that are normally essential. Instead of your fix in #2346, what we need is a README in the `test/worlds` directory to tell users not to use these files, but instead refer to the `examples` directory.

I was not aware of that. Thanks for the clarification.

To my knowledge, there is no example usage of `DopplerVelocityLogSystem` present in the `examples` directory. Do you think it would be a good idea to provide one?

**Comment by arjo129** [comment]
Yes @sauk2.  Rather than adding it to the integration tests (which dont need visuals) create a new doppler velocity log example file.

**Comment by sauk2** [comment]
> Yes @sauk2. Rather than adding it to the integration tests (which dont need visuals) create a new doppler velocity log example file.

@arjo129 Understood. Will go ahead with that. 
Should I add that README too as a part of that PR? 

If yes, then is there any additional information it should contain other than what is mentioned in the following comment?

> Hi @sauk2 , those files are meant to be used for integration tests. They don't need the `SceneBroadcaster` because the tests don't need the GUI. Adding it will unnecessarily increase test time. Most of the SDFormat files there intentionally don't have the `SceenBroadcaster` or other systems that are normally essential. Instead of your fix in #2346, what we need is a README in the `test/worlds` directory to tell users not to use these files, but instead refer to the `examples` directory.

**Comment by arjo129** [comment]
Yes thatd be great!!
